src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -246,38 +248,91 @@ snc_quota_manager::calc_node_quota_default() const {
     return default_quota;
 }
 
+void snc_quota_manager::get_or_create_quota_context(
+  std::unique_ptr<snc_quota_context>& ctx,
+  std::optional<std::string_view> client_id) {
+    if (likely(ctx)) {
+        // NB: comparing string_view to sstring might be suboptimal
+        if (likely(ctx->_client_id == client_id)) {
+            // the context is the right one
+            return;
```

Comment:
nit: if you  made the signature `std::unique_ptr<ctx> get_or_create(std::unique_ptr<ctx>)` then we could get rid of the input/output parameter and the call site would clearly be identified at what is constructing the context. but no big deal either way.